src/pages/waypoint/docs/index.tsx

```diff
@@ -0,0 +1,172 @@
+import { IconTerminalScreen16 } from '@hashicorp/flight-icons/svg-react/terminal-screen-16'
+
+// Global imports
+import waypointData from 'data/waypoint.json'
+import { ProductData } from 'types/products'
+import SidebarSidecarLayout from 'layouts/sidebar-sidecar'
+import { getStaticGenerationFunctions } from 'layouts/sidebar-sidecar/server'
+import IconTileLogo from 'components/icon-tile-logo'
+import Heading from 'components/heading'
+import Text from 'components/text'
+import CardLink from 'components/card-link'
+import ProductDocsSearch from 'views/docs-view/components/product-docs-search'
+import DocsView from 'views/docs-view'
+import CardsGridList from 'components/cards-grid-list'
+import TruncateMaxLines from 'components/truncate-max-lines'
+import devDotStyles from 'components/dev-dot-content/dev-dot-content.module.css'
+
+// Local imports
+import pageContent from './content.json'
+import s from './style.module.css'
+
+// Constants
+const basePath = 'docs'
+const baseName = 'Docs'
+const product = waypointData as ProductData
+const iconMap = {
+  box: <IconBox16 />,
+  'terminal-screen': <IconTerminalScreen16 />,
+}
+const heroCards = [
+  {
+    iconName: 'terminal-screen',
+    link: '/waypoint/commands',
+    text: 'CLI',
+  },
+  {
+    iconName: 'box',
+    link: '/waypoint/plugins',
+    text: 'Plugins',
+  },
+]
+
+interface WaypointDocsLandingProps {
+  mdxSource: MDXRemoteSerializeResult
+}
+
+// TODO: abstract the markup to a new view, such as RootDocsPathLandingView
+const WaypointDocsLanding = ({ mdxSource }: WaypointDocsLandingProps) => {
+  const { pageSubtitle } = pageContent
+
+  return (
+    <>
+      {__config.flags.enable_product_docs_search ? <ProductDocsSearch /> : null}
+      <div className={s.pageHeroWrapper}>
+        <div className={s.pageHeroText}>
+          <IconTileLogo productSlug="waypoint" />
+          <div>
+            <Heading className={s.pageTitle} level={1} size={500} weight="bold">
+              Waypoint Documentation
+            </Heading>
+            <Text className={s.pageSubtitle}>{pageSubtitle}</Text>
+          </div>
+        </div>
+        <ul className={s.heroCardList}>
+          {heroCards.map(({ iconName, link, text }) => (
+            <li key={link}>
+              <CardLink className={s.heroCardLink} href={link}>
+                {iconMap[iconName]}
+                <Text
+                  asElement="span"
+                  className={s.heroCardLinkText}
+                  size={200}
+                  weight="semibold"
+                >
+                  {text}
+                </Text>
+              </CardLink>
+            </li>
+          ))}
+        </ul>
+      </div>
+      <div className={s.waypointMDXWrapper}>
+        <DocsView mdxSource={mdxSource} hideSearch />
+      </div>
+      <div>
+        {pageContent.marketingContentBlocks.map((block) => {
+          if (block.type === 'card-grid') {
```

Comment:
💭 Thought: I found including `heading` as _part of a block_ rather than _as a separate block_ made things harder at the "build the sidecar" stage. Definitely something I started noticing _after_ building the dot-io docs landing pages (IIRC they made the same call, but I think I somewhat regret it).

The "it was hard" example: In [product-tutorials-view](https://github.com/hashicorp/dev-portal/blob/2428520b75ce77861a96d906d2329409b0364c4b/src/views/product-tutorials-view/components/product-view-content/index.tsx#L26), I found the existing block structure to be harder to work with. The idea that each block can contain nested elements felt like it made things particularly hard; but even without that, having headings _not_ be their own separate blocks felt like it made content less flexible than we really want it, and also made it more complicated to extract headings for the sidecar.

The "it felt better" example: in [product-landing-view](https://github.com/hashicorp/dev-portal/blob/2428520b75ce77861a96d906d2329409b0364c4b/src/views/product-landing/components/product-landing-blocks/index.tsx#L14), the flat array of authorable blocks make it more intuitive to write flexibly grouped content (eg multiple sets of cards "under" one heading), and it felt easier to reason about (IMO) when extracting headings, and when mapping over and dealing with CSS spacing between elements on the page.

☝️ Relatively hot take here, but wanted to note in case in sparks anything! Would mean some interesting changes, eg the concept of a "add support for description text" to the card-grid block (as mentioned in a comment below) would be re-conceptualized as supporting three simpler blocks: heading, text, and card grid. Take that with a big grain of salt, I might be off here, and I think the block structure already here in this PR is great either way 👍

Reply:
Agreed on this! It's partially why I decided to defer the Sidecar for a follow-up PR. I'm not 100% convinced either way just yet, but my mentality so far with having this block that has `title` and (upcoming) `description` properties is so that the responsibility of knowing some semantic HTML and heading element hierarchy isn't on the people that edit this page. So my goal thus far has been to enable flexibility in the content rendered but not so much flexibility that someone could accidentally break the accessibility of the page.

The main tradeoff IMO is increased complexity in the developer experience with (hopefully!) decreased complexity in the authoring experience. This is probably going to get more complicated when I work on /vault/docs next, because its content is less "flat" than Waypoint's currently. The `card-grid` in Waypoint's renders an `<h2>` heading because that's the appropriate heading given the MDX content that comes before it. On Vault however, the headings will need to be level 3 because they will be under the "Use Cases" heading which will be a level 2 heading.

<details>
<summary>Expand for /vault/docs design screenshot</summary>
<img src="https://user-images.githubusercontent.com/43934258/169310784-814d7cc0-574d-49f8-b704-35dc6efc2a9f.png" />
</details>

But anyways! I'm going to see how implementing the Vault page goes, and if it doesn't go quickly enough I will probably compromise on these attempts in the short term and we can spend appropriate time on making the authoring experience more robust later. Totally appreciate your thoughts here and I'm glad to know we're pretty much on the same page!